mh/constants.py

```diff
@@ -164,13 +165,41 @@ def make_binary_npc_greeters(is_jap=False):
     US_OFFSET = 0x180
     JP_OFFSET = 0x100
     offset = JP_OFFSET if is_jap else US_OFFSET
+
+
+    if is_jhen_active():
+        tool_shop_str = "Half-off sale!"
+        material_shop_str = "Half-off sale!"
+        event_quests_str = "The Festival of Fear\n"
+    else:
+        tool_shop_str = "Prices are normal."
+        material_shop_str = "Prices are normal."
+        event_quests_str = ""
+
+    event_quests_str += QUEST_EVENT_JUMP_FOUR_JAGGI[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_BLOOD_SPORT[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_MERCY_MISSION[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_THE_PHANTOM_URAGAAN[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_WORLD_EATER[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_WHERE_GODS_FEAR_TO_TREAD[0:40].decode('ascii').rstrip('\x00') + '\n'
+    event_quests_str += QUEST_EVENT_GREEN_EGGS[0:40].decode('ascii').rstrip('\x00')
```

Comment:
You should create a helper function which extracts the quest's name:
```python
event_quests_str = b"\n".join([
    get_quest_name(QUEST_1),
    get_quest_name(QUEST_2),
    ...
])
```

Reply:
Done

Reply:
`b"\n".join` is missing and `get_quest_name` should return `bytes`.

Reply:
Fixed